scaleway/helpers_instance.go

```diff
@@ -146,6 +146,55 @@ func reachState(ctx context.Context, instanceAPI *instance.API, zone scw.Zone, s
 	return nil
 }
 
+// prepareVolumeForAttach will make sure a volume is either not attached to a server or already attached to the given server. If volume is attached to another server, it will be stopped
+// to allow volume detachment.
+// Note: the Volume.ServerSummary may not be accurate
+func prepareVolumeForAttach(ctx context.Context, instanceAPI *instance.API, zone scw.Zone, volumeID string, serverID string) (*instance.Volume, error) {
+	res, err := instanceAPI.GetVolume(&instance.GetVolumeRequest{
+		Zone:     zone,
+		VolumeID: volumeID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return nil, err
+	}
+
+	if res.Volume.Server == nil || res.Volume.Server.ID == serverID {
+		return res.Volume, nil
+	}
+
+	defer lockLocalizedID(newZonedIDString(zone, res.Volume.Server.ID))()
```

Comment:
We should try to avoid using `lockLocalizedID` as we try to deprecate it cf https://github.com/scaleway/terraform-provider-scaleway/issues/662 @jerome-quere